### PR TITLE
Better error codes and messages

### PR DESCRIPTION
**Pull request by martonpe**
* https://github.com/sspinc/csv2avro/issues/24 Exit with 2 if parameters are missing or there was an error while processing
* Write usage and error messages to STDERR

### CODE REVIEW

File: bin/csv2avro [hunk 1]
```diff
@@ -48,11 +48,12 @@ begin
...
+  $stderr.puts option_parser
+  exit 2
 rescue Exception => e
-  puts 'Uh oh, something went wrong!'
-  puts e.message
-  puts e.backtrace.join("\n")
+  $stderr.puts 'Uh oh, something went wrong!'
+  $stderr.puts e.message
+  $stderr.puts e.backtrace.join("\n")
+  exit 2
```
Comment (gszjulcsi) [inline bin/csv2avro]: I'd exit with different error codes in each cases. That may help debug the issue later.

**Comment by gszjulcsi** [comment]
LGTM with comments

File: bin/csv2avro (hunk 1 above)
Comment (rgabo) [inline bin/csv2avro]: A typical case of `exit 1`, something went wrong is exit code 1 pretty much everywhere:

https://www.youtube.com/watch?v=7QLSRMoKKS0

**Comment by rgabo** [comment]
For reference:

http://tldp.org/LDP/abs/html/exitcodes.html

**Pull request merged**